unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3913 @@
+    l_layoutSurface.id_surface = 2;
+    mp_iviShell->bkgnd_surface_id = 1;
+
+    struct weston_surface *lp_westonSurface = (struct weston_surface *)malloc(sizeof(struct weston_surface));
+    SET_RETURN_SEQ(surface_get_weston_surface, &lp_westonSurface, 1);
+
+    surface_event_create(&mp_iviShell->surface_created, &l_layoutSurface);
+
+    EXPECT_EQ(get_id_of_surface_fake.call_count, 1);
+    EXPECT_EQ(get_properties_of_surface_fake.call_count, 1);
+    EXPECT_EQ(surface_get_weston_surface_fake.call_count, 1);
+    EXPECT_EQ(surface_add_listener_fake.call_count, 1);
+    EXPECT_EQ(wl_resource_post_event_fake.call_count, OBJECT_NUMBER);
+    EXPECT_EQ(wl_resource_post_event_fake.arg1_history[0], IVI_WM_SURFACE_CREATED);
+    EXPECT_EQ(wl_resource_post_event_fake.arg1_history[1], IVI_WM_SURFACE_CREATED);
+    EXPECT_EQ(OBJECT_NUMBER, g_SurfaceCreatedCount);
+    struct ivisurface *lp_iviSurf = (struct ivisurface*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct ivisurface, notification_list));
+    EXPECT_EQ(lp_iviSurf->shell, mp_iviShell);
+    EXPECT_EQ(lp_iviSurf->layout_surface, &l_layoutSurface);
+    EXPECT_EQ(mp_iviShell->bkgnd_surface, nullptr);
+
+    free(lp_iviSurf);
+    free(lp_westonSurface);
+}
+
+/** ================================================================================================
+ * @test_id             surface_event_create_idSameBkgnSurfaceId
+ * @brief               Test case of surface_event_create() where id_surface same bkgnd_surface_id
+ * @test_procedure Steps:
+ *                      -# Set id_surface and bkgnd_surface_id
+ *                      -# Mocking the surface_get_weston_surface() does return an object
+ *                      -# Calling the surface_event_create()
+ *                      -# Verification point:
+ *                         +# get_id_of_surface() must be called once time
+ *                         +# get_properties_of_surface() must be called once time
+ *                         +# surface_get_weston_surface() must be called once time
+ *                         +# surface_add_listener() not be called
+ *                         +# wl_resource_post_event() not be called
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, surface_event_create_idSameBkgnSurfaceId)
+{
+    struct ivi_layout_surface l_layoutSurface;
+    l_layoutSurface.id_surface = 1;
+    mp_iviShell->bkgnd_surface_id = 1;
+
+    struct weston_surface *lp_westonSurface = (struct weston_surface *)malloc(sizeof(struct weston_surface));
+    SET_RETURN_SEQ(surface_get_weston_surface, &lp_westonSurface, 1);
+
+    surface_event_create(&mp_iviShell->surface_created, &l_layoutSurface);
+
+    EXPECT_EQ(get_id_of_surface_fake.call_count, 1);
+    EXPECT_EQ(get_properties_of_surface_fake.call_count, 1);
+    EXPECT_EQ(surface_get_weston_surface_fake.call_count, 1);
```

Comment:
These verification points are redundant in my opinion, because in the surface_event_create_idDifferentBkgnSurfaceId, we have already verified these points for `surface_event_create()` function, there is no need to verify it again. The redundant verification points for a unit (e.g. function) make more effort in the future when we modify the code, then we need to modify all test cases to verify the same point.

Reply:
From my opinion, we should keep to check in each testcase, because if we just verify these condition in case surface_event_create_idDifferentBkgnSurfaceId, and when we only want to run case surface_event_create_idSameBkgnSurfaceId (for example run with gtest flag --gtest_filter), these function will not be verifed